### PR TITLE
BUG: Grouped-by column loses name when empty list of aggregations is specified

### LINKED ISSUE #32580: [BUG] Grouped-by column loses name when empty list of aggregations is specified.
```
In [53]: a = pd.DataFrame({'a': [1, 1, 2], 'b': [1, 2, 3], 'c': [1, 2, 4]})

In [54]: a.groupby('a').agg({'c': ['min']}) # name preserved in index
Out[54]:
    c
  min
a
1   1
2   4

In [55]: a.groupby('a').agg({'b': [], 'c': ['min']}) # name lost in index
Out[55]:
    c
  min
1   1
2   4
```

In the above snippet, the name of the grouped by column (`'a'`) is preserved in the former case, and lost in the latter case.

### PR DESCRIPTION
**Pull request by echozzy629**
- [ ] closes #32580
- [ ] 0 tests added / 0 passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/core/base.py [hunk 1]
```diff
@@ -438,7 +438,17 @@ def is_any_frame() -> bool:
                 # we have a dict of DataFrames
                 # return a MI DataFrame
 
-                return concat([result[k] for k in keys], keys=keys, axis=1), True
+                #return concat([result[k] for k in keys], keys=keys, axis=1), True
+                # #issue 32580: Grouped-by column loses name when empty list of aggregations is specified.
+                #Bug in the method `DataFrame.groupby` lost index, when one of the ``agg`` keys referenced an empty list (:issue:`32580`)
+                #return concat([result[k] for k in keys], keys=keys, axis=1), True
```
Comment (jbrockmendel) [inline pandas/core/base.py:444]: can you clean this up, remove commented-out code, just well-formatted comments

File: pandas/core/base.py [hunk 2]
```diff
@@ -438,7 +438,17 @@ def is_any_frame() -> bool:
...
+                keys_to_use=[k for k in keys if not result[k].empty]
+                # check: if at least one DataFrame is not empty
+                if keys_to_use !=[]:
+                    keys_to_use=keys_to_use
+                else:
+                    keys_to_use=keys_to_use
+                return(concat([result[k] for k in keys_to_use], keys=keys_to_use, axis=1), True
```
Comment (jbrockmendel) [inline pandas/core/base.py:451]: can you run `black` to handle formatting issues

**Comment by jbrockmendel** [comment]
needs a test for the resolved issue

File: doc/source/development/code_style.rst [hunk 3]
```diff
@@ -9,7 +9,7 @@ pandas code style guide
 .. contents:: Table of contents:
    :local:
 
-*pandas* follows the `PEP8 <https://www.python.org/dev/peps/pep-0008/>`_
+pandas follows the `PEP8 <https://www.python.org/dev/peps/pep-0008/>`_
```
Comment (mroeschke) [inline doc/source/development/code_style.rst:12]: Can you keep these pandas changes in another PR?

**Comment by simonjayhawkins** [comment]
@echozzy629 closing as stale. ping if you want to continue.
